src/containers/deactivateAccount/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { Alert, Button, Checkbox, Form, Typography } from 'antd';
+import React from 'react';
+import { Helmet } from 'react-helmet';
+import { connect, useDispatch } from 'react-redux';
+import { Link as RouterLink, Redirect } from 'react-router-dom';
+import { UserAuthenticationReducerState } from '../../auth/ducks/types';
+import { ContentContainer } from '../../components';
+import { C4CState } from '../../store';
+import {
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+  asyncRequestIsLoading,
+} from '../../utils/asyncRequest';
+import { requestToDeactivateAccount } from './ducks/thunks';
+import { DeactivateAccountReducerState } from './ducks/types';
+const { Title, Link } = Typography;
+
+const DeactivateAccount: React.FC<{
+  deactivateAccount: DeactivateAccountReducerState['deactivateAccount'];
+  tokens: UserAuthenticationReducerState['tokens'];
+}> = ({ deactivateAccount, tokens }) => {
+  const dispatch = useDispatch();
+  const onFinish = () => {
+    // delete the account
+    dispatch(requestToDeactivateAccount());
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Deactivate Account</title>
+        <meta name="description" content="Permanently delete your account." />
+      </Helmet>
+      <ContentContainer>
+        <Title>Deactivate Account</Title>
+        {asyncRequestIsComplete(deactivateAccount) ? (
```

Comment:
This ternary logic is a bit hard to read. What do you think about using a switch statement in an IIFE?

Reply:
I think because each of these states are mutually exclusive we can do a series of `(asyncIs___() && ...}`